docs: introduction content (#102)
svecosystem/runed: sites/docs/content/index.md

Runed is a collection of utilities for Svelte 5 that make composing powerful applications and
libraries a breeze, leveraging the power of [Svelte Runes](https://svelte.dev/blog/runes).

## Why Runed?

Svelte 5 Runes unlock immense power by providing a set of primitives that allow us to build
impressive applications and libraries with ease. However, building complex applications often
requires more than just the primitives provided by Svelte Runes.

Runed takes those primitives to the next level by providing:

- **Powerful Utilities**: A set of carefully crafted utility functions and classes that simplify
  common tasks and reduce boilerplate.
- **Collective Efforts**: We often find ourselves writing the same utility functions over and over
  again. Runed aims to provide a single source of truth for these utilities, allowing the community
  to contribute, test, and benefit from them.

> - **Consistency**: We aim to provide consistent APIs and behaviors across all utilities, so you can
>   focus on building your projects instead of constantly learning new APIs.

Reviewer: ```suggestion
- **Consistency**: A consistent set of APIs and behaviors across all utilities, so you can
  focus on building your projects instead of constantly learning new APIs.
```

- **Reactivity First**: Powered by Svelte 5's new reactivity system, Runed utilities are designed to
  handle reactive state and side effects with ease.
- **Type Safety**: Full TypeScript support to catch errors early and provide a better developer
  experience.

## Ideas and Principles

#### Embrace the Magic of Runes

Svelte Runes are a powerful new paradigm. We believe in fully embracing this concept and exploring
its potential. Our goal is to make working with Runes feel as natural and intuitive as possible.

#### Enhance, Don't Replace

> We're not here to replace Svelte's core functionality, but to enhance and extend it. Our utilities
> should feel like a natural extension of Svelte, not a separate framework.

Reviewer: ```suggestion
Runed is not here to replace Svelte's core functionality, but to enhance and extend it. Our utilities
should feel like a natural extension of Svelte, not a separate framework.
```

#### Progressive Complexity

> Simple things should be simple, complex things should be possible. We provide easy-to-use defaults
> while allowing for advanced customization when needed.

Reviewer: ```suggestion
Simple things should be simple, complex things should be possible. Runed provides easy-to-use defaults
while allowing for advanced customization when needed.
```
